lib/utils/parameter_utils.dart

```diff
@@ -8,3 +13,133 @@ bool nameConsistsOfUnderscoresOnly(FormalParameter parameter) {
+    String? correctionMessage,
+    String? uniqueName,
+    String? url,
+    ErrorSeverity? errorSeverity,
+  }) =>
+      LintCode(
+        name: name ?? this.name,
+        problemMessage: problemMessage ?? this.problemMessage,
+        correctionMessage: correctionMessage ?? this.correctionMessage,
+        uniqueName: uniqueName ?? this.uniqueName,
+        url: url ?? this.url,
+        errorSeverity: errorSeverity ?? this.errorSeverity,
+      );
+}
+
+/// Source: https://github.com/epam-cross-platform-lab/swagger-dart-code-generator/blob/master/lib/src/extensions/yaml_extensions.dart
+extension YamlMapConverter on YamlMap {
+  dynamic _convertNode(dynamic v) {
+    if (v is YamlMap) {
+      return v.toMap();
+    } else if (v is YamlList) {
+      final list = <dynamic>[];
+      for (final e in v) {
+        list.add(_convertNode(e));
+      }
+      return list;
+    } else {
+      return v;
+    }
+  }
+
+  /// Converts [YamlMap] to [Map<String, dynamic>]
+  Map<String, dynamic> toMap() {
+    final map = <String, dynamic>{};
+    nodes.forEach(
+      (k, v) => map[(k as YamlScalar).value.toString()] = _convertNode(v.value),
+    );
+    return map;
+  }
+}
+
+/// Determines whether the file should be skipped using Glob patterns
+///
+/// See https://pub.dev/packages/glob
+bool shouldSkipFile({
+  required List<String> includeGlobs,
+  required List<String> excludeGlobs,
+  required String path,
+  String? rootPath,
+}) {
+  if (includeGlobs.isEmpty && excludeGlobs.isEmpty) {
+    return false;
+  }
+
+  final bool isIncludeOnly = excludeGlobs.isEmpty && includeGlobs.isNotEmpty;
+  final bool isExcludeOnly = includeGlobs.isEmpty && excludeGlobs.isNotEmpty;
+  final pathNormalized = normalizePath(
+    path,
+    rootPath != null ? normalizePath(rootPath) : null,
+  );
+  if (isIncludeOnly) {
+    return _isFileNotIncluded(includeGlobs, path);
+  }
+
+  if (isExcludeOnly) {
+    return _isFileExcluded(includeGlobs, pathNormalized);
+  }
+
+  final isFileNotIncluded = _isFileNotIncluded(includeGlobs, pathNormalized);
+  final isFileExcluded = _isFileExcluded(excludeGlobs, pathNormalized);
+  return isFileNotIncluded || isFileExcluded;
+}
+
+bool _isFileExcluded(List<String> excludes, String path) {
+  final hasMatch =
+      excludes.map(Glob.new).toList().any((glob) => glob.matches(path));
+  return hasMatch;
+}
+
+bool _isFileIncluded(List<String> includes, String path) {
+  final hasMatch =
+      includes.map(Glob.new).toList().any((glob) => glob.matches(path));
+  return hasMatch;
+}
+
+bool _isFileNotIncluded(List<String> includes, String path) {
+  return !_isFileIncluded(includes, path);
+}
+
+/// Normalizes the path using posix style and
+/// replaces backslashes with forward slashes
+String normalizePath(String path, [String? root]) {
+  // Remove drive letter from path
+  final String subpath = _removeDriveLetter(path);
+
+  return p.posix.relative(subpath.replaceAll(r'\', '/'), from: root);
+}
```

Comment:
Please, use `Uri.normalizePath` instead.

Reply:
Alright, I've also renamed it so it is more accurate to what it returns. For context, the globs fail when the drive letter is in the path so using relative is needed, and it doesn't make sense to match against folders above the project.
```dart
/// Converts path to relative using posix style and
/// replaces backslashes with forward slashes
String relativePath(String path, [String? root]) {
  final uriNormlizedPath = p.toUri(path).normalizePath().path;
  final uriNormlizedRoot =
      root != null ? p.toUri(root).normalizePath().path : null;

  final relative = p.posix.relative(uriNormlizedPath, from: uriNormlizedRoot);
  return relative;
}
```